sql/parser/Makefile

```diff
@@ -1,5 +1,6 @@
-.PHONY: all
-all: sql.go keywords.go reserved_keywords.go
+TARGETS = sql.go keywords.go reserved_keywords.go
+.PHONY: all $(TARGETS)
```

Comment:
Yeah, I suppose it makes sense to always regenerate `sql.go` when make is invoked. We've taken the style of placing `.PHONY` declarations immediately in front of the targets they apply to (see the top-level `Makefile` for examples). So:

```
.PHONY: sql.go
sql.go:
   ...

.PHONY: keywords.go
keywords.go:
   ...
```

Also, remove the dependency of `sql.go` on `sql.y`. Now that `sql.go` is a phony target that isn't necessary and slightly confusing. Ditto for the dependencies for `keywords.go` and `reserved_keywords.go`.

Note we could have also fixed this by making `sql.go` depend on `sql_union.sh` and `$(which yacc)`. Seems safer to always rebuild, though.